From 95df14db22d7c473bcb52469445b7e8b10bb6fdb Mon Sep 17 00:00:00 2001
From: bloodshot <jdroque@gmail.com>
Date: Wed, 18 Jun 2014 22:20:10 -0400
Subject: [PATCH] Add missing GameMode used by FakePlayers in Bukkit API. Fixes
 #1233

FakePlayers do not set their GameMode which results in using the default
'NOT_SET(-1)' GameMode. Since this does not exist in Bukkit's API, any
references from plugins to player.getGameMode() will be null. This commit
simply adds the missing gamemode.

diff --git a/src/main/java/org/bukkit/GameMode.java b/src/main/java/org/bukkit/GameMode.java
index 803944ea..9b6d220f 100644
--- a/src/main/java/org/bukkit/GameMode.java
+++ b/src/main/java/org/bukkit/GameMode.java
@@ -11,6 +11,12 @@ import com.google.common.collect.Maps;
  * have
  */
 public enum GameMode {
+    // Cauldron start - FakePlayers do not set their gametype and use the default
+    /**
+     * Default mode
+     */
+    NOT_SET(-1),
+    // Cauldron end
     /**
      * Creative mode may fly, build instantly, become invulnerable and create
      * free items.
-- 
2.16.2.windows.1

